test/general/sycl_iterator.pass.cpp

```diff
@@ -3082,257 +3565,296 @@ struct test_set_intersection
-        auto host_first3 = get_host_pointer(first3);
+        host_res.retrieve_data();
         auto nres_expect =
-            ::std::set_union(host_first1, host_first1 + na, host_first2, host_first2 + nb, res_expect) - res_expect;
-        EXPECT_EQ_N(host_first3, res_expect, nres_expect, "wrong effect from set_union a, b");
+            ::std::set_union(host_keys.get(), host_keys.get() + na, host_vals.get(), host_vals.get() + nb, res_expect) - res_expect;
+        EXPECT_EQ_N(host_res.get(), res_expect, nres_expect, "wrong effect from set_union a, b");
     }
 };
 
-struct test_set_symmetric_difference
+DEFINE_TEST(test_set_symmetric_difference)
 {
+    DEFINE_TEST_CONSTRUCTOR(test_set_symmetric_difference)
+
     template <typename Policy, typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
     void
     operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Iterator2 last2, Iterator3 first3,
-               Iterator3 last3, Size /* n */)
+               Iterator3 last3, Size n)
     {
-        auto host_first1 = get_host_pointer(first1);
-        auto host_first2 = get_host_pointer(first2);
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, get_size(n));
+        TestDataTransfer<UDTKind::eVals, Size> host_vals(*this, get_size(n));
+        TestDataTransfer<UDTKind::eRes, Size>  host_res (*this, get_size(n));
+
         last1 = first1 + na;
         last2 = first2 + nb;
-        ::std::copy(a, a + na, host_first1);
-        ::std::copy(b, b + nb, host_first2);
+
+        ::std::copy(a, a + na, host_keys.get());
+        ::std::copy(b, b + nb, host_vals.get());
+        host_keys.update_data(na);
+        host_vals.update_data(nb);
 
         last3 = ::std::set_symmetric_difference(make_new_policy<new_kernel_name<Policy, 0>>(exec), first1, last1,
-                                              first2, last2, first3);
+                                                first2, last2, first3);
 #if _PSTL_SYCL_TEST_USM
         exec.queue().wait_and_throw();
 #endif
         int res_expect[na + nb];
-        auto host_first3 = get_host_pointer(first3);
-        auto nres_expect =
-            ::std::set_symmetric_difference(host_first1, host_first1 + na, host_first2, host_first2 + nb, res_expect) -
-            res_expect;
-        EXPECT_EQ_N(host_first3, res_expect, nres_expect, "wrong effect from set_symmetric_difference a, b");
+        retrieve_data(host_keys, host_vals, host_res);
+        auto nres_expect = ::std::set_symmetric_difference(host_keys.get(), host_keys.get() + na, host_vals.get(),
+                                                           host_vals.get() + nb, res_expect) -
+                           res_expect;
+        EXPECT_EQ_N(host_res.get(), res_expect, nres_expect, "wrong effect from set_symmetric_difference a, b");
     }
 };
 #endif
 
-std::int32_t
-main()
-{
 #if TEST_DPCPP_BACKEND_PRESENT
+template <sycl::usm::alloc alloc_type>
+void
+test_usm()
```

Comment:
1) should we renamed to 'test_usm_and_buffer'
2) in a caller of 'test_usm_and_buffer' should be added "//TODO: There is the over-testing here - each algorithm is run with sycl::buffer as well. So, in case of a couple of 'test_usm_and_buffer' call we get double-testing case with sycl::buffer."

Reply:
should we renamed to 'test_usm_and_buffer' - done.